kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/auth/SimpleAclAuthorizer.java

```diff
@@ -152,8 +158,26 @@ public CompletableFuture<Boolean> canProduceAsync(KafkaPrincipal principal, Reso
     public CompletableFuture<Boolean> canConsumeAsync(KafkaPrincipal principal, Resource resource) {
         checkResourceType(resource, ResourceType.TOPIC);
         TopicName topicName = TopicName.get(resource.getName());
+        if (forceCheckGroupId && StringUtils.isBlank(principal.getGroupId())) {
+            return CompletableFuture.completedFuture(false);
+        }
         return authorizationService.canConsumeAsync(
-                topicName, principal.getName(), principal.getAuthenticationData(), "");
+                topicName, principal.getName(), principal.getAuthenticationData(), principal.getGroupId());
+    }
+
+    @Override
+    public CompletableFuture<Boolean> canDescribeConsumerGroup(KafkaPrincipal principal, Resource resource) {
+        if (forceCheckGroupId) {
+            if (StringUtils.isBlank(principal.getGroupId())) {
+                return CompletableFuture.completedFuture(false);
+            }
+            boolean isSameGroup = Objects.equals(principal.getGroupId(), resource.getName());
+            if (log.isDebugEnabled()) {
+                log.debug("Principal [{}] for resource [{}] isSameGroup [{}]", principal, resource, isSameGroup);
+            }
+            return CompletableFuture.completedFuture(isSameGroup);
+        }
+        return CompletableFuture.completedFuture(true);
```

Comment:
Not a request change, but it would be better if you use guard clause.

```java
        if (!forceCheckGroupId) {
            return CompletableFuture.completedFuture(true);
        }
        // ...
```